From 95b4bcb4da2ad740879f054ba36840468ca0c756 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 26 Aug 2014 14:12:30 -0400
Subject: [ethernet] igb: use ethtool_cmd_speed_set helper to set ethtool speed value

Message-id: <1409062359-13063-73-git-send-email-sassmann@redhat.com>
Patchwork-id: 89180
O-Subject: [RHEL7.1 PATCH 72/81] net: use ethtool_cmd_speed_set helper to set ethtool speed value
Bugzilla: 1091121
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Neil Horman <nhorman@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Pirko <jiri@resnulli.us>

Signed-off-by: Jiri Pirko <jiri@resnulli.us>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit d4f3cd49d2800dc037724efa9b33c485a1cc23d3)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/igb/igb_ethtool.c b/drivers/net/ethernet/intel/igb/igb_ethtool.c
index d1d0803..4d154e2 100644
--- a/drivers/net/ethernet/intel/igb/igb_ethtool.c
+++ b/drivers/net/ethernet/intel/igb/igb_ethtool.c
@@ -141,6 +141,7 @@ static int igb_get_settings(struct net_device *netdev, struct ethtool_cmd *ecmd)
  struct e1000_dev_spec_82575 *dev_spec = &hw->dev_spec._82575;
  struct e1000_sfp_flags *eth_flags = &dev_spec->eth_flags;
  u32 status;
+ u32 speed;
 
  status = rd32(E1000_STATUS);
  if (hw->phy.media_type == e1000_media_type_copper) {
@@ -215,13 +216,13 @@ static int igb_get_settings(struct net_device *netdev, struct ethtool_cmd *ecmd)
  if (status & E1000_STATUS_LU) {
   if ((status & E1000_STATUS_2P5_SKU) &&
       !(status & E1000_STATUS_2P5_SKU_OVER)) {
-   ecmd->speed = SPEED_2500;
+   speed = SPEED_2500;
   } else if (status & E1000_STATUS_SPEED_1000) {
-   ecmd->speed = SPEED_1000;
+   speed = SPEED_1000;
   } else if (status & E1000_STATUS_SPEED_100) {
-   ecmd->speed = SPEED_100;
+   speed = SPEED_100;
   } else {
-   ecmd->speed = SPEED_10;
+   speed = SPEED_10;
   }
   if ((status & E1000_STATUS_FD) ||
       hw->phy.media_type != e1000_media_type_copper)
@@ -229,9 +230,10 @@ static int igb_get_settings(struct net_device *netdev, struct ethtool_cmd *ecmd)
   else
    ecmd->duplex = DUPLEX_HALF;
  } else {
-  ecmd->speed = SPEED_UNKNOWN;
+  speed = SPEED_UNKNOWN;
   ecmd->duplex = DUPLEX_UNKNOWN;
  }
+ ethtool_cmd_speed_set(ecmd, speed);
  if ((hw->phy.media_type == e1000_media_type_fiber) ||
      hw->mac.autoneg)
   ecmd->autoneg = AUTONEG_ENABLE;
-- 
1.7.1